LibOS/shim/src/bookkeep/shim_handle.c

```diff
@@ -628,13 +628,15 @@ static struct shim_handle_map * __enlarge_handle_map
     if (size <= map->fd_size)
         return map;
 
-    size_t new_size = size * sizeof(*map->map);
-    struct shim_fd_handle ** new_map = realloc(map->map, new_size);
+    struct shim_fd_handle ** new_map = calloc(size, sizeof(new_map[0]));
+
     if (!new_map)
         return NULL;
 
-    size_t copy_size = sizeof(*map->map) * map->fd_size;
-    memset(map->map + copy_size, 0, new_size - copy_size);
+    memcpy(new_map, map->map, map->fd_size * sizeof(new_map[0]));
+    memset(new_map + map->fd_size, 0,
+           (size - map->fd_size) * sizeof(new_map[0]));
+    map->map = new_map;
```

Comment:
The old value of `map->map` leaks here. This is why I added realloc - it simplifies the code and makes it less leak-prone. Why did you remove it?